fluentinterface/src/main/java/com/iluwatar/fluentinterface/fluentiterable/FluentIterable.java

```diff
@@ -0,0 +1,81 @@
+package com.iluwatar.fluentinterface.fluentiterable;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+/**
+ * The FluentIterable is a more convenient implementation of the common iterable interface based
+ * on the fluent interface design pattern.
+ * This interface defines common operations, but
+ * doesn't aim to be complete. It was inspired by Guava's com.google.common.collect.FluentIterable.
+ * @param <TYPE> is the class of objects the iterable contains
+ */
+public interface FluentIterable<TYPE> extends Iterable<TYPE> {
+
+    /**
+     * Filters the iteration with the given predicate.
+     * @param predicate the condition to test with for the filtering. If the test
+     *                  is negative, the tested object is removed by the iterator.
+     * @return a filtered FluentIterable
+     */
+    FluentIterable<TYPE> filter(Predicate<? super TYPE> predicate);
+
+    /**
+     * Uses the Iterable interface's forEach method to apply a given function
+     * for each object of the iterator. This is a terminating operation.
+     */
+    void forEachDo(Consumer<? super TYPE> action);
+
+    /**
+     * Evaluates the iteration and returns the first element. This is a terminating operation.
+     * @return the first element after the iteration is evaluated
+     */
+    Optional<TYPE> first();
+
+    /**
```

Comment:
Javadoc: `Evaluates` is not a correct representation in interface because in case of Lazy iterable the evaluation will only occur on termination operation.

Javadoc: `iteration` in javadoc should be changed to `Iterable`